docs/source/app-dev/bindings-scala/index.rst

```diff
@@ -0,0 +1,66 @@
+.. Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Scala bindings
+##############
+
+.. toctree::
+  :hidden:
+
+The Scala bindings is a client implementation of the *Ledger API*
+based on `Akka Streams API <https://doc.akka.io/docs/akka/current/stream/index.html>`_.
+
+Introduction
+************
+
+The Scala bindings library lets you write applications that connect to the Digital Asset distributed ledger using Scala programming language.
+
+There are two main components:
+
+- The Akka Streams based API
+    The API to send commands to the ledger and receive transactions back.
+
+- Scala codegen
+    DAML to Scala code generator. Generates Scala classes from DAML models. The generated Scala code provides a type safe way of creating contracts (create command) and exercising contract choices (exercise command).
+
+The reader must be familiar with:
+
+- :doc:`DAML language </daml/reference/index>`
+- :doc:`Ledger API </app-dev/grpc/index>`
+- `Akka Streams API <https://doc.akka.io/docs/akka/current/stream/index.html>`_
+- `Scala programming language <https://www.scala-lang.org>`_
+
+Getting started with SBT
+************************
+
+Setting up dependencies in ``build.sbt``:
+
+.. literalinclude:: ./code-snippets/quickstart-scala/build.sbt
+   :start-after: // <doc-ref:dependencies>
+   :end-before: // </doc-ref:dependencies>
+
+There are two modules in the example below:
+
+- ``scala-codegen``
+    This modules contains generated Scala ADTs.
+- ``application``
+    This is the application code that makes use of the generated Scala ADTs.
```

Comment:
Is this just needed to get your quickstart example working, or will everyone need this?

Reply:
yep, everyone will need it. This is a recommended way of setting up an application that uses scala codegen. Two modules, one module that contains generated classes and another module for the application code.